src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -388,18 +389,68 @@ function ScatterplotViz(props: VisualizationProps) {
       ? allData?.series
       : allData.facets.find(({ data }) => data.series.length > 0)?.data.series;
 
+    // logic for setting markerColor correctly
+    // find raw legend label (excluding No data as well)
+    const legendLabel = legendData
+      ?.filter(
+        (data) =>
+          !data.name?.includes(', Smoothed mean') &&
+          !data.name?.includes(', 95% Confidence interval') &&
+          !data.name?.includes(', Best fit') &&
+          !data.name?.includes('No data')
+      )
+      .map((data) => data.name);
+
+    // construct a kind of a lookup table
+    const legendLabelColor = legendLabel?.map((label, index) => {
+      return {
+        label: label,
+        color: ColorPaletteDefault[index],
+      };
+    });
+
     return legendData != null
-      ? legendData.map((data: XYPlotDataSeries, index: number) => {
+      ? // the name 'dataItem' is used inside the map() to distinguish from the global 'data' variable
+        legendData.map((dataItem: XYPlotDataSeries, index: number) => {
           return {
-            label: data.name != null ? data.name : '',
-            // TO-do: need a way to appropriately make marker info
-            // scatter plot has defined mode - still need to distinguish CI, No data, etc.
-            marker: data.mode != null ? data.mode : '',
-            // set marker colors
-            markerColor: 'markerColor',
+            label: dataItem.name != null ? dataItem.name : '',
+            // maing marker info appropriately
+            marker:
+              dataItem.mode != null
+                ? dataItem.name === 'No data'
+                  ? 'x'
+                  : dataItem.mode === 'markers'
+                  ? 'circle'
+                  : dataItem.mode === 'lines'
+                  ? 'line'
+                  : ''
+                : dataItem.fill != null && dataItem.fill === 'tozeroy'
```

Comment:
Maybe something like

```suggestion
                : dataItem?.fill === 'tozeroy'
```

Reply:
Yep!